### DISCUSSION TITLE
Wav file from php script / wavesurfer-multitrack

### DISCUSSION THREAD

**Discussion by fabiantorress** [katspaugh/wavesurfer.js#3237] [Q&A]
Hi, I am trying to use wavesurfer-multitrack in Nextjs. I am sending my wav file data from my php backend but I am having problems to use the wav file in my client.

The url must I pass to the multitrack options like this or there is another way?    
```
 {
      id: 1,
      draggable: false,
      startPosition: 0,
      fadeInEnd: 8,
      fadeOutStart: 18,
      volume: 0.8,
      options: {
        waveColor: "hsl(161, 87%, 49%)",
        progressColor: "hsl(161, 87%, 20%)",
      },
###       url: "http://xx.xx.xx.xx/api-monitor/cdr_audio_files.php",
    },
```
I am a bit lost, thanks in advance

**Comment by katspaugh** [comment]
Yes, `url` is the correct property.

**Reply by fabiantorress** [reply]
And there is a way to show a loading state while the component is fetching the wav file?
And in my php backend is sending the wav file like this: <?php
header("Access-Control-Allow-Origin: *");
header("Access-Control-Allow-Headers: *");
header('Access-Control-Allow-Methods: *');
$file_path = '/path/to/file/wavFile.wav'; 
$file_name = 'wavFile.wav';

if (file_exists($file_path)) {
    header('Content-Description: File Transfer');
    header('Content-Type: application/octet-stream');
    header('Content-Disposition: attachment; filename="' . $file_name . '"');
    header('Expires: 0');
    header('Cache-Control: must-revalidate');
    header('Pragma: public');
    header('Content-Length: ' . filesize($file_path));
    readfile($file_path);
    exit;
} else {
    echo 'File not found.';
} Is there something wrong with my php api that is why is not rendering the multitrack?

**Reply by katspaugh** [reply]
The content type should be `audio/wav`.

Generally, serving static files with a scripting language is a bad idea. There are many web servers out there that do this out of the box, much faster and much more compliant with standards. E.g. AWS S3, Nginx. This is especially true for audio files because they need to accept byte ranges when seeking.
